packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -139,6 +139,8 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
         } else {
           this._onItemSurfaceClick(ev, index);
         }
+      } else {
+        selection.setAllSelected(false);
```

Comment:
Right. That's the design. Moving to a non selectable entity should not be clearing your selection.